src/targets/pubDev.ts

```diff
@@ -0,0 +1,208 @@
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   * If there's required software missing, raises an error.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug(
+      'Checking if Dart CLI is available: ',
+      this.pubDevConfig.dartCliPath
+    );
+    checkExecutableIsPresent(this.pubDevConfig.dartCliPath);
+  }
+
+  /**
+   * Uploads all files to pub.dev using Dart CLI.
+   *
+   * @param version New version to be released
+   * @param revision Git commit SHA to be published
+   */
+  public async publish(_version: string, revision: string): Promise<any> {
+    await this.createCredentialsFile();
+
+    await withTempDir(
+      async directory => {
+        await this.cloneRepository(this.githubRepo, revision, directory);
+        await forEachChained(this.pubDevConfig.packages, async pkg =>
+          this.publishPackage(directory, pkg)
+        );
+      },
+      true,
+      'craft-pub-dev-'
+    );
+  }
+
+  public async createCredentialsFile(): Promise<void> {
+    const credentialsFilePath = this.getCredentialsFilePath();
+    const content = {
+      accessToken: this.pubDevConfig.PUBDEV_ACCESS_TOKEN,
+      refreshToken: this.pubDevConfig.PUBDEV_REFRESH_TOKEN,
+      tokenEndpoint: 'https://accounts.google.com/o/oauth2/token',
+      scopes: ['openid', 'https://www.googleapis.com/auth/userinfo.email'],
+      expiration: 1645564942000, // Expiration date is required, but irrelevant
+    };
+
+    // Store credentials file only if they doesn't exist
+    try {
+      await fsPromises.access(credentialsFilePath, constants.F_OK);
+      this.logger.warn('Credentials file already exists. Skipping creation.');
+    } catch {
+      await fsPromises.writeFile(credentialsFilePath, JSON.stringify(content));
+      this.logger.info('Credentials file created.');
+    }
+  }
+
+  private getCredentialsFilePath(): string {
+    const currentPlatform = platform();
+    switch (currentPlatform) {
+      case 'darwin':
+        return `${homedir()}/Library/Application Support/dart/pub-credentials.json`;
+      case 'linux':
+        return `${homedir()}/.config/dart/pub-credentials.json`;
+      default:
+        throw new Error(`Unsupported platform: ${currentPlatform}`);
+    }
+  }
+
+  /**
+   * Clones a repository.
+   *
+   * @param config Git configuration specifying the repository to clone.
+   * @param revision The commit SHA that should be checked out after the clone.
+   * @param directory The directory to clone into.
+   */
+  public async cloneRepository(
+    config: GitHubGlobalConfig,
+    revision: string,
+    directory: string
+  ): Promise<any> {
+    const { owner, repo } = config;
+    const git = simpleGit(directory);
+    const url = `https://github.com/${owner}/${repo}.git`;
+
+    this.logger.info(`Cloning ${owner}/${repo} into ${directory}`);
+    await git.clone(url, directory);
```

Comment:
The functionality of cloning a repo already exists in various places: [`crates` target](https://github.com/getsentry/craft/blob/e0fe1b134850c74099368cd12ec17e1fa9cc9b06/src/targets/crates.ts#L301), [`registry` target](https://github.com/getsentry/craft/blob/e0fe1b134850c74099368cd12ec17e1fa9cc9b06/src/targets/registry.ts#L429) and [`upm` target](https://github.com/getsentry/craft/blob/e0fe1b134850c74099368cd12ec17e1fa9cc9b06/src/targets/upm.ts#L125), for example. Can we extract it to avoid duplicated code?

Reply:
I was thinking about this, but only `crates` performs clone+checkout. `registry` and `upm` does completely other actions, so in the end we'd abstract already encapsulated in `SimpleGit` actions.